slsDetectorSoftware/src/SharedMemory.h

```diff
@@ -64,8 +64,23 @@ template <typename T> class SharedMemory {
             unmapSharedMemory();
     }
 
-    T *operator()() { return shared_struct; }
-    const T *operator()() const { return shared_struct; }
+    inline std::string getNoShmAccessMessage() const {
```

Comment:
Could/should we make this a private function since the message is only used for the exception? I also think inline is not needed, usually the optimizer does it's thing anyway (even with inline it doesn't always get inlined)


```cpp
//alternative to avoid allocating the intermediate string. 
const char* noShmErrorMessage()
```

Reply:
makes sense. will do